labshare-client/src/backend/database/database.ts

```diff
@@ -53,26 +72,18 @@ export function getModelForRole(role: string): Optional<Model<IUserCommon>> {
 }
 
 export async function getUser(filter: any): Promise<Optional<IUserCommon>> {
-    let models: Model<any>[] = [UserLabResearch, UserLabDiag, UserVolunteer]
-    let matches = []
+    return (await UserCommon.findOne(filter).exec()) ?? undefined
+}
+
+export async function getUserOrAdmin(filter: any): Promise<Optional<IUserCommon>> {
+    let models: Model<any>[] = [UserCommon, UserAdmin]
 
     for (let i of models) {
-        try {
-            let match = await i.findOne(filter).exec()
-            if (match) {
-                matches.push(match)
-            }
-        } catch {
-            return undefined
+        let tmp = await i.findOne(filter)
+        if (tmp) {
+            return tmp
         }
     }
-
-    if (matches.length > 1) {
-        throw new Error("More than one result found.")
-    } else if (matches.length == 0) {
-        return undefined
-    } else {
-        return matches[0]
-    }
-}
-
+    
+    return undefined
```

Comment:
Don't get it...

Reply:
Sorry for being a bit too obscure. and a bit too quick on the draw, because I see I made a mistake. The whole method here is equivalent to

```typescript
export async function getUserOrAdmin(filter: {...some approrpiate find type}): Promise<...> {
 return [UserCommon, UserAdmin].map(async model => await model.findOne(filter)).filter(Boolean)[0];
}
```

That might be a bit too terse, and it fires off two requests each time.

On second thought, I think the following approach might be better:

```typescript
export async function getUserOrAdmin(...): ... {
return (await UserCommon.findOne(filter)) || (await UserAdmin.findOne(filter));
}
```

That is both more expressive and less wasteful than either of our ideas so far.